wgpu/tests/buffer.rs

```diff
@@ -91,18 +97,100 @@ fn test_empty_buffer_range(ctx: &TestingContext, buffer_size: u64, label: &str)
 
     b1.unmap();
 
-    for _ in  0..10 {
+    for _ in 0..10 {
         ctx.device.poll(wgpu::MaintainBase::Poll);
     }
 }
 
 #[test]
+#[ignore]
 fn empty_buffer() {
-    initialize_test(
-        TestParameters::default(),
-        |ctx| {
-            test_empty_buffer_range(&ctx, 2048, "regular buffer");
-            test_empty_buffer_range(&ctx, 0, "zero-sized buffer");
+    // TODO: Currently wgpu does not accept empty buffer slices, which
+    // is what test is about.
+    initialize_test(TestParameters::default(), |ctx| {
+        test_empty_buffer_range(&ctx, 2048, "regular buffer");
+        test_empty_buffer_range(&ctx, 0, "zero-sized buffer");
+    })
+}
+
+#[test]
+fn test_map_offset() {
+    initialize_test(TestParameters::default(), |ctx| {
+        // This test writes 16 bytes at the beginning of buffer mapped mapped with
+        // an offset of 32 bytes. Then the buffer is copied into another buffer that
+        // is read back and we check that the written bytes are correctly placed at
+        // offset 32..48.
+        // The goal is to check that get_mapped_range did not accidentally double-counts
+        // the mapped offset.
+
+        let status = Arc::new(AtomicBool::new(false));
+
+        let write_buf = ctx.device.create_buffer(&wgpu::BufferDescriptor {
+            label: None,
+            size: 256,
+            usage: wgpu::BufferUsages::MAP_WRITE | wgpu::BufferUsages::COPY_SRC,
+            mapped_at_creation: false,
+        });
+        let read_buf = ctx.device.create_buffer(&wgpu::BufferDescriptor {
+            label: None,
+            size: 256,
+            usage: wgpu::BufferUsages::MAP_READ | wgpu::BufferUsages::COPY_DST,
+            mapped_at_creation: false,
+        });
+
+        let done = status.clone();
+        write_buf
+            .slice(32..)
+            .map_async(wgpu::MapMode::Write, move |result| {
+                assert!(result.is_ok());
+                done.store(true, Ordering::SeqCst);
+            });
+
+        while !status.load(Ordering::SeqCst) {
+            ctx.device.poll(wgpu::MaintainBase::Poll);
+        }
```

Comment:
What's the point of this vs just waiting?

Reply:
Are you referring to using `MaintainBase::Wait` instead of `Poll` ? I don't know, both appear to work, changing it to Wait.